ts/features/wallet/onboarding/privative/screens/search/LoadPrivativeSearch.tsx

```diff
@@ -1,22 +1,70 @@
-import { View } from "native-base";
 import * as React from "react";
 import { connect } from "react-redux";
 import { Dispatch } from "redux";
+import { isNone } from "fp-ts/lib/Option";
+import I18n from "../../../../../../i18n";
 import { GlobalState } from "../../../../../../store/reducers/types";
+import {
+  searchUserPrivative,
+  walletAddPrivativeCancel
+} from "../../store/actions";
+import {
+  onboardingSearchedPrivativeSelector,
+  SearchedPrivativeData
+} from "../../store/reducers/searchedPrivative";
+import { onboardingPrivativeFoundIsError } from "../../store/reducers/foundPrivative";
+import { useHardwareBackButton } from "../../../../../bonus/bonusVacanze/components/hooks/useHardwareBackButton";
+import { LoadingErrorComponent } from "../../../../../bonus/bonusVacanze/components/loadingErrorScreen/LoadingErrorComponent";
+import { WithTestID } from "../../../../../../types/WithTestID";
+import { showToast } from "../../../../../../utils/showToast";
+import { mixpanelTrack } from "../../../../../../mixpanel";
+import { toPrivativeQuery } from "./SearchPrivativeCardScreen";
 
-type Props = ReturnType<typeof mapDispatchToProps> &
-  ReturnType<typeof mapStateToProps>;
+type Props = WithTestID<
+  ReturnType<typeof mapDispatchToProps> & ReturnType<typeof mapStateToProps>
+>;
 
 /**
  * Loading screen while search the privative card
  * @param props
  * @constructor
  */
-const LoadPrivativeSearch = (_: Props): React.ReactElement => <View />;
+const LoadPrivativeSearch = (props: Props): React.ReactElement | null => {
+  useHardwareBackButton(() => {
+    if (!props.isLoading) {
+      props.cancel();
+    }
+    return true;
+  });
 
-const mapDispatchToProps = (_: Dispatch) => ({});
+  const privativeQueryParam = toPrivativeQuery(props.privativeSelected);
+  if (isNone(privativeQueryParam)) {
+    showToast(I18n.t("global.genericError"), "danger");
+    void mixpanelTrack("PRIVATIVE_NO_QUERY_PARAMS_ERROR");
```

Comment:
Could you add the description of the event in https://docs.google.com/spreadsheets/d/1bXNRcTDKU-5l2RqCcwSbT7Y1M9QQTWtS_f-3-AIqSYk/edit#gid=1909779257&range=A109 ?